src/Microsoft.Framework.Configuration.EnvironmentVariables/EnvironmentVariables/EnvironmentVariablesConfigurationSource.cs

```diff
@@ -76,22 +76,26 @@ private static IEnumerable<DictionaryEntry> AzureEnvToAppEnv(DictionaryEntry ent
             {
                 prefix = CustomPrefix;
             }
-            else
+
+            key = key.Substring(prefix.Length).Replace("__", ":");
```

Comment:
`Substring(0)` returns the whole string - so isn't that still fine? Maybe I've misunderstood this.